### PR TITLE
Add pinia

### PR DESCRIPTION
**Pull request by isSerge**
- update Quasar and related dependencies
- add Pinia store for state management
- decouple state from presentation (move state-related logic from components to store)
- add Jest test framework and some tests for store (will add to GH workflow once #279 is merged)

### CODE REVIEW

File: src/pages/Dashboard.vue [hunk 1]
```diff
@@ -2,153 +2,83 @@
...
-      if (config.nodeName !== "") {
-        await this.$client.startNode(config.plot.location, config.nodeName)
-      } else {
-        util.errorLogger("DASHBOARD | node name was empty when tried to start node")
-      }
-      util.infoLogger("DASHBOARD | starting farmer")
-      const farmerStarted = await this.$client.startFarming(config.plot.location, config.plot.sizeGB)
-      if (!farmerStarted) {
-        util.errorLogger("DASHBOARD | Farmer start error!")
-      }
```
Comment (ozgunozerk) [inline src/pages/Dashboard.vue:87]: why the error handling is abandoned?

Comment (isSerge) [reply src/pages/Dashboard.vue:87]: error logging is still in place - it was moved to the store, error handling will be implemented within #253

**Review by ozgunozerk — APPROVED** [review]
I really liked the new organization of the code. Thanks @isSerge

**Review by nazar-pc — COMMENTED** [review]
Skimmed though PR, but I don't have a lot of context to what is happening.
Overall it feels like this could have been a few PRs and I wish commit message explained what bug was fixed exactly.
Not approving because I don't think I can review it properly, but I also have nothing against merging it.

**Comment by isSerge** [comment]
@nazar-pc there were some bugs as a result of refactoring

**Pull request merged**